client/apps/edit/components/header2/index.jsx

```diff
@@ -0,0 +1,188 @@
+  }
+
+  contentIsComplete = () => {
+    const { article } = this.props
+    return article.get('title') && article.get('title').length
+  }
+
+  displayIsComplete = () => {
+    const { article } = this.props
+    return article.get('thumbnail_title') && article.get('thumbnail_image')
+  }
+
+  render () {
+    const { article, actions, channel, user } = this.props
+    const { activeSection, isDeleting } = this.props.edit
+
+    return (
+      <div className='EditHeader'>
+
+        <div className='EditHeader__left'>
+          <div className='EditHeader__tabs'>
+            <button
+              className='avant-garde-button check'
+              onClick={() => this.toggleSection('content')}
+              data-active={activeSection === 'content'}
+            >
+              <span>Content</span>
+              <Icon
+                className='icon'
+                name='check'
+                color={this.contentIsComplete() ? 'limegreen' : '#ccc'}
+              />
+            </button>
+
+            <button
+              className='avant-garde-button check'
+              onClick={() => this.toggleSection('display')}
+              data-active={activeSection === 'display'}
+            >
+              <span>Display</span>
+              <Icon
+                className='icon'
+                name='check'
+                color={this.displayIsComplete() ? 'limegreen' : '#ccc'}
+              />
+            </button>
+
+            {user.type === 'Admin' &&
+              <button
+                className='avant-garde-button'
+                onClick={() => this.toggleSection('admin')}
+                data-active={activeSection === 'admin'}
+              >
+                Admin
+              </button>
+            }
+          </div>
+
+          <div>
+            <button
+              className='avant-garde-button publish'
+              data-disabled={!this.isPublishable()}
+              onClick={this.onPublish}
+            >
+              {this.getPublishText()}
+            </button>
+
+            {channel.get('type') === 'editorial' &&
+              <button
+                className='avant-garde-button autolink'
+              >
+                Auto-link
+              </button>
+            }
+          </div>
+        </div>
+
+        <div className='EditHeader__right'>
+          <button
+            className='avant-garde-button'
+            style={{border: 'none'}}
+            onClick={this.onDelete}
+          >
+            {isDeleting ? 'Deleting...' : 'Delete'}
+          </button>
+
+          <button
+            className='avant-garde-button'
+            style={{color: this.getSaveColor()}}
+            onClick={() => article.get('published')
+              ? article.trigger('savePublished')
+              : actions.saveArticle(article)
+            }>
+            {this.getSaveText()}
+          </button>
+
+          <a href={`${sd.FORCE_URL}/article/${article.get('slug')}`} target='_blank'>
```

Comment:
There's a model method for this! You'll want to double check but I think it's `article.getFullSlug()`

Reply:
Hot tip!  The function isn't being any used anywhere except now here. Updated to use underscore.string's slugify, which is what we use in API (some special characters were showing in slug).
Also found the 'contentIsComplete' and 'displayIsComplete' existed on the model.